Sources/Yams/Representer.swift

```diff
@@ -192,38 +200,71 @@ extension Optional: NodeRepresentable {
 }
 
 extension Decimal: ScalarRepresentable {
-    public func represented() throws -> Node {
-        return Node(description)
+    public func represented() -> Node.Scalar {
+        return .init(description)
     }
 }
 
 extension URL: ScalarRepresentable {
-    public func represented() throws -> Node {
-        return Node(absoluteString)
+    public func represented() -> Node.Scalar {
+        return .init(absoluteString)
+    }
+}
+
+extension String: ScalarRepresentable {
+    public func represented() -> Node.Scalar {
+        return .init(self)
     }
 }
 
 /// MARK: - ScalarRepresentableCustomizedForCodable
 
-public protocol ScalarRepresentableCustomizedForCodable: ScalarRepresentable {
-    func representedForCodable() -> Node
+public protocol YAMLEncodable: Encodable {
+    func box() -> Node
 }
 
-extension Date: ScalarRepresentableCustomizedForCodable {
-    public func representedForCodable() -> Node {
+extension YAMLEncodable where Self: ScalarRepresentable {
+    public func box() -> Node {
+        return .scalar(represented())
+    }
+}
+
+extension Bool: YAMLEncodable {}
+extension Data: YAMLEncodable {}
+extension Decimal: YAMLEncodable {}
+extension Int: YAMLEncodable {}
+extension Int8: YAMLEncodable {}
+extension Int16: YAMLEncodable {}
+extension Int32: YAMLEncodable {}
+extension Int64: YAMLEncodable {}
+extension UInt: YAMLEncodable {}
+extension UInt8: YAMLEncodable {}
+extension UInt16: YAMLEncodable {}
+extension UInt32: YAMLEncodable {}
+extension UInt64: YAMLEncodable {}
+extension URL: YAMLEncodable {}
+extension String: YAMLEncodable {}
```

Comment:
Can we do `extension StringProtocol: YAMLEncodable`?

Reply:
Same with `BinaryInteger`.
```
Yams/Sources/Yams/Representer.swift:250:1: error: extension of protocol 'StringProtocol' cannot have an inheritance clause
extension StringProtocol: YAMLEncodable {
^                         ~~~~~~~~~~~~~
```